eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/beaconv2/BeaconServiceV2.java

```diff
@@ -0,0 +1,356 @@
+
+@Service
+public class BeaconServiceV2 {
+
+    private static final String ORGANIZATION_ID = "EMBL-EBI-EVA";
+
+    private static final String ORGANIZATION_NAME = "European Variation Archive (EMBL-EBI)";
+
+    private static final String ORGANIZATION_DESCRIPTION = "EMBL-EBI makes the world's public biological data freely" +
+            " available to the scientific community via a range of services and tools, performs basic research" +
+            " and provides professional training in bioinformatics. The European Variation Archive is an " +
+            "open-access" + " database of all types of genetic variation data from all species.";
+
+    private static final String ORGANIZATION_ADDRESS = "Wellcome Genome Campus, Hinxton, Cambridgeshire, CB10 1SD, " +
+            "United Kingdom";
+
+    private static final String ORGANIZATION_WELCOME_URL = "www.ebi.ac.uk/eva";
+
+    private static final String ORGANIZATION_CONTACT_URL = "contactUrlString";
+
+    private static final String ORGANIZATION_LOGO_URL = "www.ebi.ac.uk/eva/img/eva_logo.png";
+
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @Autowired
+    private VariantSourceService variantSourceService;
+
+    public BeaconServiceV2() {
+    }
+
+    public BeaconImpl getBeacon() {
+        BeaconImpl beacon = new BeaconImpl();
+        beacon.setDatasets(getAllBeaconDatasets());
+        return beacon;
+    }
+
+    private List<BeaconDataset> getAllBeaconDatasets() {
+        List<BeaconDataset> beaconDatasets = new ArrayList<>();
+        beaconDatasets.addAll(getBeaconDatasetsPerDatabase("hsapiens_grch37", "randomDescription", "GRCh37",
+                "updatedatetime", "externalurl"));
+        beaconDatasets.addAll(getBeaconDatasetsPerDatabase("hsapiens_grch38", "randomDescription", "GRCh38",
+                "updatedatetime", "externalurl"));
+        return beaconDatasets;
+    }
+
+    private List<BeaconDataset> getBeaconDatasetsPerDatabase(String db, String description, String assemblyId,
+                                                             String updateDateTime, String externalUrl) {
+        List<BeaconDataset> beaconDatasets = new ArrayList<>();
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(db));
+        List<VariantSource> variantSources = variantSourceService.findAllVariantSourcesForBeacon();
+        variantSources.forEach(
+                variantSource -> beaconDatasets.add(
+                        new BeaconDataset().id(variantSource.getStudyId())
+                                .name(variantSource.getStudyName())
+                                .description(description)
+                                .assemblyId(assemblyId)
+                                .createDateTime(variantSource.getDate() == null ? null :
+                                        variantSource.getDate().toString())
```

Comment:
Given the way our internal processes work, this would be the update instead of the create date time.

Reply:
Fixed :+1: